cogs/roles.py

```diff
@@ -32,132 +32,89 @@ def __init__(self, bot):
         self.bot = bot
         self.roles = self.bot.config.roles
 
-    @commands.command()
-    async def pronoun(self, ctx, transaction, pronoun):
+    async def toggleRole(self, ctx, transaction, requestedRole, category):
         """
-        Self-assign a pronoun role to a user. 
+        Assigs a single role to a user with no checks from a category of roles
         """
         member = ctx.message.author
+        roles = self.roles.get(category).split(", ")
 
-        # normalize pronoun
-        pronoun = pronoun.title()
-
-        pronouns = self.roles.get("pronouns").split(", ")
-
-        if (transaction == "add") and (pronoun in pronouns):
-            role = discord.utils.get(ctx.guild.roles, name=pronoun)
+        # normalize user input to title case
+        requestedRole = requestedRole.title()
 
+        if (transaction == "add") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
             await member.add_roles(role, reason="Self Requested")
-        elif (transaction == "remove") and (pronoun in pronouns):
-            role = discord.utils.get(ctx.guild.roles, name=pronoun)
+        elif (transaction == "remove") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
             await member.remove_roles(role, reason="Self Requested")
-
         else:
-            await ctx.send("Error, that role doesn't exist")
+            await ctx.send("Error, that role doesn't exist in this category")
+        return
 
-    @commands.command()
-    async def field(self, ctx, transaction, field):
+    async def toggleRestrictedRole(self, ctx, transaction, requestedRole,
```

Comment:
I kind of like `toggleUniqueRole` a bit better because it seems clearer to me what the purpose of the function is. Up to you though